mods/persistence/modules/trade_beacons/trade_beacons.dm

```diff
@@ -0,0 +1,114 @@
+/atom/proc/get_adjacent_trade_beacons()
+	var/list/adjacent_trade_beacons = list()
+	var/atom/target = src
+	if(!istype(src, /turf))
+		target = get_turf(src)
+	var/obj/ob = target.get_owning_overmap_object()
+	if(ob)
+		var/turf/T = get_turf(ob)
+		if(T)
+			for(var/obj/beacon in SStrade_beacons.all_trade_beacons)
+				var/turf/beacon_turf = get_turf(beacon)
+				if(beacon_turf && beacon_turf.Adjacent(T))
+					adjacent_trade_beacons |= beacon
+	return adjacent_trade_beacons
+
+/obj/effect/overmap/trade_beacon
+	name = "Trade Beacon"
+	var/list/possible_imports = list() // structure is list(/datum/beacon_import/example = 100) list(type = percentage chance of appearing).
+	var/list/possible_exports = list()
+	var/list/active_imports = list()
+	var/list/active_exports = list()
+	icon = 'icons/obj/supplybeacon.dmi'
+	icon_state = "beacon_active"
+	var/datum/extension/network_device/trade_controller/linked_controller
+	var/datum/money_account/beacon_account
+	var/start_x = 0
+	var/start_y = 0
+
+
+/obj/effect/overmap/trade_beacon/Destroy()
+	forceMove(null)
```

Comment:
You don't need to move it to nullspace tbh. The base Destroy() proc does it for you. It's not a blocking change though.
```suggestion
```